### PR TITLE
Replace android run.sh, build.sh, and make-release.sh with fastlane

### PR DESCRIPTION
**Pull request by nicknovitski**

### CODE REVIEW

**Review by ide — CHANGES_REQUESTED** [review]
See comments about whether the prod_release lane should build an APK or not.

File: android/shell.nix [hunk 1]
```diff
@@ -7,20 +7,26 @@ let
 
 in
 
-mkShell {
+mkShell rec {
 
   LANG="en_US.UTF-8";
   JAVA_HOME=openjdk8;
 
+  ANDROID_SDK_ROOT=builtins.getEnv("HOME") + (if stdenv.isDarwin then "/Library/Android/sdk" else "/Android/Sdk");
```
[the remark this answers is not in the document]
Comment (nicknovitski) [reply android/shell.nix]: Yeah.  Weird, isn't it?  I'll add a comment.

File: .circleci/config.yml [hunk 2]
```diff
@@ -252,6 +251,7 @@ jobs:
       - setup
       - restore_cache:
```
[the remark this answers is not in the document]
Comment (nicknovitski) [reply .circleci/config.yml]: I mistakenly thought that the restored apk would act as a build cache, and the prod_release step would mostly skip over the build, but I just determined locally that is not the case, so never mind.

We do want to keep a build job for building release builds and one releasing release builds.  Given that, there's no reason not to reuse the output of one as the input to the other.  I just erroneously believed I was still doing that.

File: fastlane/Fastfile [hunk 3]
```diff
@@ -126,16 +126,34 @@ platform :android do
...
+    )
+  end
+
   lane :prod_release do
     build_gradle = File.read("../android/app/build.gradle")
 
     verify_changelog_exists(version_code: build_gradle.match(/versionCode (\d+)/)[1])
     verify_upload_to_staging(version_name: build_gradle.match(/versionName '([\d\.]+)'/)[1])
 
+    build(build_type: "Release")
```
[the remark this answers is not in the document]
Comment (nicknovitski) [reply fastlane/Fastfile]: I've backed out this line.

**Review by ide — APPROVED** [review]

**Pull request merged**